src/Type/Row.purs

```diff
@@ -70,3 +74,17 @@ instance listToRowCons
   :: ( ListToRow tail tailRow
      , RowCons label ty tailRow row )
   => ListToRow (Cons label ty tail) row
+
+-- | Ensure that every field in a row has the same type.
+class Homogeneous (row :: # Type) fieldType
```

Comment:
I though I'd asked this already but can't see it now.

Is `| row -> fieldType` missing here?

Reply:
Yes I think so